docs/data/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,214 @@
+2. `ERROR`: The submission did not go through due to an error in Stellar-Core. Take a look at the attached error message for more details, modify your transaction if necessary, and resubmit.
+3. `TRY_AGAIN_LATER`: This indicates that the Stellar-Core instance is currently unable to process submission of this particular transaction. Clients should wait for sometime before resubmitting the transaction. This could happen due to different reasons:
+   - There is another transaction from same source account in memory
+   - It has been rejected due to too low fee and has been resubmitted too soon
+   - The transaction has resource consumption higher than the network limits allow: clients should wait for sometime before retrying it again.
+
+<CodeExample>
+
+```js
+let server = sdk.Server("https://horizon-testnet.stellar.org");
+let contractId = "CA3D5KRYM6CB7OWQ6TWYRR3Z4T7GNZLKERYNZGGA5SOAOPIFY6YQGAXE";
+let contract = new StellarSdk.Contract(contractId);
+
+// Right now, this is just the default fee for this example.
+const fee = StellarSdk.BASE_FEE;
+
+let transaction = new StellarSdk.TransactionBuilder(account, { fee })
+  .setNetworkPassphrase(StellarSdk.Networks.TESTNET)
+  .setTimeout(30) // valid for the next 30s
+  // Add an operation to call increment() on the contract
+  .addOperation(contract.call("increment"))
+  .build();
+
+// Sign this transaction with the secret key
+// NOTE: signing is transaction is network specific. Test network transactions
+// won't work in the public network. To switch networks, use the Network object
+// as explained above (look for StellarSdk.Network).
+let sourceKeypair = StellarSdk.Keypair.fromSecret(sourceSecretKey);
+transaction.sign(sourceKeypair);
+
+server.submitAsyncTransaction(transaction).then((result) => {
+  console.log("hash:", result.hash);
+  console.log("status:", result.tx_status);
+  console.log("errorResultXdr:", result.error_result_xdr);
+});
+```
+
+</CodeExample>
+
+#### Synchronous Transaction Submission
+
+Due to the blocking nature of this endpoint, things are a little different compared to the asynchronous strategy. There are 3 possible scenarios that clients can encounter:
+
+1. The submission is successful and Horizon returns the transaction response back. This is the happy path and clients do not need to do anything but wait for Horizon's response.
+2. Stellar-Core sends back an `ERROR` response from the submission. Clients should consult the attached error message and retry the submission again.
+3. Timeouts: Horizon may respond with a `504` HTTP code. This response is not an error but a warning that your transaction hasn't been accepted by the network yet. There could be many possible reasons for the timeout, the most common of which is network congestion, but it could also be due to other transient issues.
```

Comment:
```suggestion
3. Timeouts: Horizon may respond with a `504` HTTP code. This response is not an error but a warning that your transaction hasn't been accepted by the network yet, so Horizon doesn't know what status to report. There could be many possible reasons for the timeout, the most common of which is network congestion, but it could also be due to other transient issues.
```